pkg/services/accesscontrol/resourcepermissions/store.go

```diff
@@ -703,3 +723,80 @@ func managedPermission(action, resource string, resourceID, resourceAttribute st
 		Scope:  accesscontrol.Scope(resource, resourceAttribute, resourceID),
 	}
 }
+
+/*
+ACTION SETS
+Stores actionsets IN MEMORY
+*/
+// - To grant access to users/teams/basic roles during runtime, users will go through the managed permissions view and access-control will be enforced based on their permissions.
+// ActionSet is a struct that represents a set of actions that can be performed on a resource.
+// An example of an action set is "folders:edit" which represents the actions that can be performed on a datasource with the uid of 1.
+/*
+```go
+actionSet := &ActionSet{
+	Resource:   "folders",
+	Permission: "edit",
+	Actions:    []string{"folders:read", "folders:write", "dashboards:read", "dashboards:write"},
+}`
+*/
+type ActionSet struct {
+	Action     string   `json:"action"`
+	Resource   string   `json:"resource"`
+	Permission string   `json:"permission"`
+	Actions    []string `json:"actions"`
+}
+
+// InMemoryActionSets is an in-memory implementation of the ActionSetService.
+type InMemoryActionSets struct {
+	log        log.Logger
+	actionSets map[string]*ActionSet
+}
+
+// NewInMemoryActionSets returns a new instance of InMemoryActionSetService.
+func NewInMemoryActionSets(log log.Logger) *InMemoryActionSets {
+	return &InMemoryActionSets{
+		actionSets: make(map[string]*ActionSet),
+		log:        log,
+	}
+}
+
+// GetActionSet returns the action set for the given action.
+func (s *InMemoryActionSets) GetActionSet(actionName string) *ActionSet {
+	actionSet, ok := s.actionSets[actionName]
+	if !ok {
+		return nil
+	}
+	return actionSet
+}
+
+func (s *InMemoryActionSets) StoreActionSet(resource, permission string, actions []string) {
+	s.log.Debug("storing action set\n")
+	name, err := s.CreateActionSetName(resource, permission)
+	if err != nil {
+		s.log.Error("error creating action set", "error", err)
+	}
+	actionSet := &ActionSet{
+		// folders:edit
+		Action:     name,
+		Resource:   resource,
+		Permission: permission,
+		Actions:    actions,
+	}
+	// TODO: Do we only store the actions, or all of the information about the action set
+	s.actionSets[actionSet.Action] = actionSet
```

Comment:
Resource and permission can be implied through the name, right (as in, name = "resource:permission"), so I think we can only store the actions.